Solutions/Branwen/GUI.cs

```diff
@@ -9,324 +9,460 @@
+						worksheetPart.Worksheet = new Worksheet(new SheetData());
+						SheetData sheetData = worksheetPart.Worksheet.GetFirstChild<SheetData>();
+						Sheet sheet = new Sheet();
+						sheet.Id = spreadsheetDocument.WorkbookPart.GetIdOfPart(worksheetPart);
+						sheet.Name = topLevelDirectories[i].Name;
+						sheet.SheetId = (UInt32)Convert.ToInt32(i + 1);
+						sheets.Append(sheet);
+						WriteDirectoryToWorksheet(RunInventory(topLevelDirectories[i], MediaDriveNumberTextBox.Text), sheetData);
+					}
+					spreadsheetDocument.Close();
+
+					#endregion
+				}
+				MessageBox.Show("DONE! Files Inventoried:  " + fileCount);
+			}
+			catch (Exception ex)
+			{
+				MessageBox.Show("Failed to Inventory:" + Environment.NewLine + ex.Message);
+			}
+			SelectAndRunInventoryButton.Enabled = true;
+			SelectAndRunInventoryButton.Text = "Select Inventory Directory";
+			UseDBCheckBox.Enabled = true;
+			SetEnabledControls();
+		}
+
+		/// <summary>
+		/// Truncates the Database to clear all items
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="e"></param>
+		private void WipeDbButton_Click(object sender, EventArgs e)
+		{
+			SelectAndRunInventoryButton.Enabled = false;
+			SelectAndRunInventoryButton.Text = "Working";
+			WipeDbButton.Enabled = false;
+			WipeDbButton.Text = "...";
+			MediaDriveNumberTextBox.Enabled = false;
+			UseDBCheckBox.Enabled = false;
+			ExportFileCheckBox.Enabled = false;
+
+			try
+			{
+				MySqlConnection mySqlConnection;
+				mySqlConnection = new MySqlConnection("server=box654.bluehost.com;user=lbkstud1_smedia;password=#sucK_my_d1ck;database=lbkstud1_SaurutobiMedia;");
+				mySqlConnection.Open();
+
+				string deleteStatement = "TRUNCATE TABLE SaurutobiMediaPaths; TRUNCATE TABLE SaurutobiMediaFiles;";
+				MySqlCommand deleteCommand = new MySqlCommand(deleteStatement, mySqlConnection);
+				deleteCommand.ExecuteNonQuery();
+			}
+			catch (Exception ex)
+			{
+				MessageBox.Show("Failed to Wipe:" + Environment.NewLine + ex.Message);
+			}
+			SelectAndRunInventoryButton.Enabled = true;
+			SelectAndRunInventoryButton.Text = "Select Inventory Directory";
+			UseDBCheckBox.Enabled = true;
+			SetEnabledControls();
+		}
+
+		/// <summary>
+		/// Switches DB-related controls Enabled property based on if it's checked
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="e"></param>
+		private void UseDBCheckBox_CheckedChanged(object sender, EventArgs e)
+		{
+			SetEnabledControls();
+		}
+
+		#endregion
+
+		#region Database Methods
+
+		/// <summary>
+		/// Writes all files in the Directory to the Databbase
+		/// </summary>
+		/// <param name="files"></param>
+		private void WriteDirectoryToDatabase(IEnumerable<BranwenFileInfo> files, MySqlConnection mySqlConnection)
+		{
+			string insertFilesStatement = "INSERT INTO SaurutobiMediaFiles (FileName, Extension, Size, MediaType, MediaDrive) VALUES ";

```

Comment:
this stuff I'm not going to go to extreme measures to adjust. I think I'm
already escaping quotes by replacing them with doublequotes. Maybe c# has a
Clean string method or something

On Thu, Dec 17, 2015 at 11:10 PM, Alex Ayerdi <notifications@github.com>
wrote:

> In Solutions/Branwen/GUI.cs
> <https://github.com/LbKStudios/Branwen/pull/5#discussion_r47999686>:
>
> > +		private void UseDBCheckBox_CheckedChanged(object sender, EventArgs e)
> > +		{
> > +			SetEnabledControls();
> > +		}
> > +
> > +		#endregion
> > +
> > +		#region Database Methods
> > +
> > +		/// <summary>
> > +		/// Writes all files in the Directory to the Databbase
> > +		/// </summary>
> > +		/// <param name="files"></param>
> > +		private void WriteDirectoryToDatabase(IEnumerable<BranwenFileInfo> files, MySqlConnection mySqlConnection)
> > +		{
> > +			string insertFilesStatement = "INSERT INTO SaurutobiMediaFiles (FileName, Extension, Size, MediaType, MediaDrive) VALUES ";
>
> All of this is also super sql injectable. I could literally have a sql
> statement as a file's name and do whatever I want to you database. Again,
> no need to change it if you don't expect people to use it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LbKStudios/Branwen/pull/5/files#r47999686>.
>